CocktailRecipes/CocktailRecipesTests/NetworkManagerTests.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  NetworkManagerTests.swift
+//  CocktailRecipesTests
+//
+//  Created by Jaison Vieira on 7/19/20.
+//  Copyright © 2020 Eric Rado. All rights reserved.
+//
+
+import XCTest
+@testable import CocktailRecipes
+
+class NetworkManagerTests: XCTestCase {
+    
+    func testRequestObjectIsSuccessful() {
+        let data = makeDictionaryData()
+        
+        let session = MockSession(response: (data, URLResponse: nil, error: nil))
+        let networkManager = NetworkManager(session: session)
+        
+        let expectation = self.expectation(description: "Send mock request")
+        
+        let completion: (Result<[String: String], Error>) -> Void = { result in
+            switch result {
+            case .failure(let error):
+                XCTAssertNil(error)
+            case .success(let response):
+                XCTAssertEqual(["lorem": "ipsum"], response)
+            }
+            
+            expectation.fulfill()
+        }
+        
+        networkManager.request(MockEndpoint(), completion: completion)
+        waitForExpectations(timeout: 1.0, handler: nil)
+    }
+    
+    func testRequestObjectListIsSuccessful() {
+        let data = makeArrayData()
+        
+        let session = MockSession(response: (data, URLResponse: nil, error: nil))
```

Comment:
Why do we need a MockSession, instead of just using URLSession.shared?

Reply:
I see why you need it, to simulate a failed network request 👍

Reply:
Because we need to override the func ```dataTask``` to be able to test.